team.md

```diff
@@ -0,0 +1,239 @@
+#### Create table team(tid, country, totalmatch, win,loss). Insert 8 team detail. Create trigger to verify totalmatch is not 0. export data into csv file. Read file using reader object and print record with winning %age.
+
+
+```python
+import sqlite3 as sq
+```
+
+
+```python
+con=sq.connect("D:/23BCA296/sqlite-tools-win-x64-3460000/team.db")
+```
+
+
+```python
+cur=con.cursor()
+```
+
+
+```python
+q="create table team(tid, country, totalmatch, win,loss)"
+```
+
+
+```python
+cur.execute(q)
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d422c0>
+
+
+
+
+```python
+con.commit
+```
+
+
+
+
+    <function Connection.commit()>
+
+
+
+
+```python
+con.close
+```
+
+
+
+
+    <function Connection.close()>
+
+
+
+### insert record
+
+
+```python
+cur=con.cursor()
+```
+
+
+```python
+cur.execute("insert into team values(1, 'USA', 10, 7, 3)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(2, 'Canada', 12, 8, 4)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(3, 'Mexico', 11, 6, 5)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(4, 'Brazil', 13, 9, 4)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(5, 'Argentina', 10, 5, 5)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(6, 'Germany', 12, 8, 4)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(7, 'France', 11, 7, 4)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+cur.execute("insert into team values(8, 'Italy', 10, 6, 4)")
+```
+
+
+
+
+    <sqlite3.Cursor at 0x2aef0d41340>
+
+
+
+
+```python
+con.commit()
+```
+
+### create trigger
+
+
+```python
+cur.execute("create trigger verify_totalmatch BEFORE INSERT  ON team\
+           begin\
+                 select case\
+                 when new.totalmatch = 0 THEN\
+                 raise(abort,'Total matches cannot be 0')\
+                 end;\
+                 end;\
+        ")
+```
+
```

Comment:
explain create trigger syntax .

Reply:
**This is the syntax for creating a trigger**
cur.execute("create trigger verify_totalmatch BEFORE INSERT  ON team\
           begin\
                 select case\
                 when new.totalmatch = 0 THEN\
                 raise(abort,'Total matches cannot be 0')\
                 end;\
                 end;\
        ")

Reply:
sorry you can't understand my question please explain your trigger syntax

Reply:
cur.execute("create trigger verify_totalmatch BEFORE INSERT ON team ..."): This line creates a new trigger named 
begin ... end;: This is the trigger body, which defines the actions to be taken when the trigger is executed.
select case .. end;: This is a conditional statement that checks the value of new.totalmatch.
when new.totalmatch = 0 THEN raise(abort, 'Total matches cannot be 0'): If the totalmatch value is 0, the trigger will raise an error with the message "Total matches cannot be 0", aborting the insert operation.